ParameterDataApi.yaml

```diff
@@ -88,122 +144,79 @@ components:
           description: The success message provided back to the client
       required:
         - message
-        
-    ColumnDefinition:
-      type: array
-      items:
-        $ref: '#/components/schemas/ColumnElementDefinition'
-          
-    ColumnElementDefinition:
-      type: object
-      description: The column elements      
-      properties:
-        
-        name: 
-          description: The name of the column
-          type: string
-          pattern: '^[a-zA-Z0-9]{2,}'
-          
-        dataType:
-          $ref: '#/components/schemas/DataType'
-                   
-      required:
-        - name
-        - dataType
 
-    DataDefinition:
-      type: array
-      items: 
-        $ref: '#/components/schemas/DataElement'
-        
-    DataElement:
+    DataSample:
       type: object
+      description: A definition of a set of parameter data having a single discrete timestamp
       properties:
+      
         timestamp:
-          description: the timestamp of the sample
           type: string
           format: date-time
-            
-        fields:
-          type: array
-          items:
-            $ref: '#/components/schemas/ValueSpec'
-            
+          description: The timestamp that applies to the measurements in the data field (DataSample)
+          
+        data: 
+          description: A row of one or more data values, associated with a single discrete timestamp. Key/value pairs should be used, including units in the key name. The fields within this object should always follow a consistent order from row to row given a combination of messageType and version
+          type: object
+          additionalProperties: true    
+        
       required:
         - timestamp
-        - fields
+        - data    
 
-    DataType:
-      type: string
-      enum:
-        - NUMBER
-        - INTEGER
-        - STRING
-        
     HeaderDefinition:
       type: object
       properties: 
-        startTime:
-          description: The start time for the time series
+      
+        componentUuid:
           type: string
-          format: date-time
-        
-        endTime: 
-          description: The end time for the time series
+          format: uuid
+          description: The unique identifier of the component within FRAIHMWORK that the parameter data associates with
+      
+        messageType:
+          type: string          
+          description: The type of the message, dictated by the client for delineating individual message formats
+
+        messageVersion:
           type: string
-          format: date-time
+          description: The version string for the parameter data messageType. If the client changes the data fields within a message stream, they should increment the version accordingly to distinguish message format along with the messageType
```

Comment:
Should semver be used/mentioned here?

Reply:
No. We're not constraining to the client that they must use semver.